### PR TITLE
chore: accessibility on dropdown button

### PR DESCRIPTION
**Pull request by Chisomchima**
Implements [LIBS-561](https://dhis2.atlassian.net/browse/LIBS-561)
---

### Description
 Implemented [LIBS-561](https://dhis2.atlassian.net/browse/LIBS-561) by updating the drowpdownButton component to close on click of the escape button. 

---

### Known issues

-   [ ] None

[LIBS-561]: https://dhis2.atlassian.net/browse/LIBS-561?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[LIBS-561]: https://dhis2.atlassian.net/browse/LIBS-561?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: components/button/src/dropdown-button/__tests__/dropdown-button.test.js [hunk 1]
```diff
@@ -43,6 +44,28 @@ describe('<DropdownButton>', () => {
...
+                    expect(queryByText(componentText)).toBeInTheDocument()
+                })
+
+                fireEvent.keyDown(document, { key: 'Escape' })
+
+                await waitFor(() => {
+                    expect(queryByText(componentText)).not.toBeInTheDocument()
+                })
+            })
         })
```
Comment (Birkbjo) [inline components/button/src/dropdown-button/__tests__/dropdown-button.test.js:69]: Could we have a test that wraps this button in a modal, and make sure that only the dropdown is closed when esc is pressed?

**Review by kabaros — CHANGES_REQUESTED** [review]
the escape functionality works well, I also tested it with a Modal @Birkbjo 

the question I have is that this fixes one part of [the requirements](https://dhis2.atlassian.net/browse/LIBS-561) - closing with ESC button. But the other part `Similar to changes to Select, a menu opened by a DropdownButton should receive focus when the button is triggered.`

dropdown-accessibility

**Comment by Chisomchima** [comment]
thanks for the feedback @kabaros and @Birkbjo I have added another test for the escape functionality of the dropdown inside a modal.

File: components/button/src/dropdown-button/__tests__/dropdown-button.test.js [hunk 2]
```diff
@@ -43,6 +45,55 @@ describe('<DropdownButton>', () => {
...
+                fireEvent.click(toggleButton)
+
+                await waitFor(() => {
+                    expect(queryByText(componentText)).toBeInTheDocument()
+                })
+
+                fireEvent.keyDown(document, { key: 'Escape' })
+
+                await waitFor(() => {
+                    expect(queryByText(componentText)).not.toBeInTheDocument()
```
Comment (Birkbjo) [inline components/button/src/dropdown-button/__tests__/dropdown-button.test.js:94]: Could we also verify that the modal is not closed here?
Looks, great!

**Comment by Chisomchima** [comment]
> @Chisomchima you still haven't handled this requirement from the ticket, right?
> 
> > Similar to changes to Select, a menu opened by a DropdownButton should receive focus when the button is triggered.

I Have written some tests to cover @Birkbjo concerns but I'll have to move this requirement to this jira [ticket](https://dhis2.atlassian.net/browse/LIBS-611)

**Comment by kabaros** [comment]
Ok - can you please make sure the ticket is in the current sprint, and has the accessibility epic as a parent.

@Birkbjo would you approve the ticket please if you're happy with it.

**Review by Birkbjo — APPROVED** [review]

**Review by kabaros — APPROVED** [review]

**Pull request merged**